neptune/internal/api_clients/hosted_api_clients/migration_switch_leaderboard_api_client_proxy.py

```diff
@@ -0,0 +1,240 @@
+#
+# Copyright (c) 2021, Neptune Labs Sp. z o.o.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+
+import logging
+import threading
+from typing import Dict, TYPE_CHECKING
+
+from neptune.model import ChannelWithLastValue
+
+from neptune.backend import LeaderboardApiClient
+
+from neptune.exceptions import ProjectMigratedToNewStructure
+from neptune.internal.api_clients.hosted_api_clients.hosted_leaderboard_api_client import \
+    HostedNeptuneLeaderboardApiClient
+
+if TYPE_CHECKING:
+    from neptune.internal.api_clients import HostedNeptuneBackendApiClient
+
+_logger = logging.getLogger(__name__)
+
+
+# pylint: disable=protected-access
+def with_migration_handling(func):
+    def wrapper(self: 'MigrationSwitchLeaderboardApiClientProxy', *args, **kwargs):
```

Comment:
And add [`wraps`](https://www.geeksforgeeks.org/python-functools-wraps-function/) just in case.